src/AspNet.Security.OAuth.Apple/Internal/DefaultAppleClientSecretGenerator.cs

```diff
@@ -133,7 +130,7 @@ private SigningCredentials CreateSigningCredentials(string keyId, ECDsa algorith
 
             // Use a custom CryptoProviderFactory so that keys are not cached and then disposed of, see below:
             // https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/1302
-            return new SigningCredentials(key, SecurityAlgorithms.EcdsaSha256Signature)
+            return new SigningCredentials(key, SecurityAlgorithms.EcdsaSha256)
```

Comment:
This change is because when `JwtSecurityTokenHandler` was used, its internals include some code to coerce algorithm names to be those supported for JWTs:

  * [Mapping table](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/b45a5b4cf4e918ca2b2c12ff1951f1be7b592bac/src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs#L82)
  * [Usage](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/b45a5b4cf4e918ca2b2c12ff1951f1be7b592bac/src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs#L551)
  * [Coercion](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/b45a5b4cf4e918ca2b2c12ff1951f1be7b592bac/src/System.IdentityModel.Tokens.Jwt/JwtHeader.cs#L101-L102)

`JsonWebTokenHandler` does not do that, and [uses the algorithm name verbatim](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/b45a5b4cf4e918ca2b2c12ff1951f1be7b592bac/src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs#L147), which was causing the wrong algorithm name to be sent to Apple, causing it to be rejected.